spec/language/for_spec.rb

```diff
@@ -0,0 +1,192 @@
+require_relative '../spec_helper'
+require_relative 'fixtures/for_scope'
+
+# for name[, name]... in expr [do]
+#   body
+# end
+describe "The for expression" do
+  it "iterates over an Enumerable passing each element to the block" do
+    j = 0
+    for i in 1..3
+      j += i
+    end
+    j.should == 6
+  end
+
+  # NATFIXME: Support Prism::ImplicitRestNode
+  xit "iterates over a list of arrays and destructures with empty comma" do
+    #for i, in [[1,2]]
+      #i.should == 1
+    #end
+  end
```

Comment:
I've written all compile time errors with `xit` block, unless we could run some code that doesn't pass the test. This way the compile time errors still show up as skipped specs.